compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -85,17 +99,25 @@ pub(super) fn generate_coverage_spans(
     let mut insert = |bcb| {
         bcb_has_mappings.insert(bcb);
     };
-    for &BcbMapping { kind, span: _ } in &mappings {
+    let mut test_vector_bitmap_bytes = 0;
+    for BcbMapping { kind, span: _ } in &mappings {
         match kind {
-            BcbMappingKind::Code(bcb) => insert(bcb),
-            BcbMappingKind::Branch { true_bcb, false_bcb } => {
-                insert(true_bcb);
-                insert(false_bcb);
+            BcbMappingKind::Code(bcb) => insert(*bcb),
+            BcbMappingKind::Branch { true_bcb, false_bcb, .. } => {
+                insert(*true_bcb);
+                insert(*false_bcb);
```

Comment:
Here we should match on `*kind`, and then add `ref` to any fields that need to be taken by reference. That way we don't have to add `*` to the individual fields in lots of places.

Reply:
Thanks for reviewing!
I had tried to this. But after BcbMappingKind::Decision with a BTreeSet was implemented the BcbMappingKind can not be Copy any more. As a result all fields are reference here. Is there any way to get the Copy fields by value from a referenced struct?

Reply:
If you do `match *kind` here, it should just work.

Because all of the fields you use are `Copy`, it doesn't matter that there are some non-copy fields in the `..`.

If you did need to use a non-copy field like `end_bcbs`, you could write `ref end_bcbs` in the pattern, and it would work.